joomla/joomla-cms
# [5.0] Implement Schema.org support for Menu Items

### Summary of Changes
Implements same Article and Contact Schema.org parameters and parsing into JSON-LD for Menu Items. This allows extensions and other custom pages to leverage Schema.org JSON-LD without needing any additional development.


### Testing Instructions
1. Edit, publish, and save "Content - Joomla" plugin
2. Edit, publish, and save "System - Schema.org" plugin
3. Create or edit a new menu item
4. Configure and save data under the Schema tab as you would with an Article
5. Navigate to that menu item on frontend
6. Press F12 to open your browsers developer tools
7. Review the <head> to make sure your JSON-LD Schema data is present

This is dependent on the menu id being explicitly loaded. So with SEO off that would be Itemid in the URL. Schema data is also cached by default so if you accessed the menu item before saving your schema data be sure to clear your cache or turn off caching of schema data.


### Actual result BEFORE applying this Pull Request
No Schema tab in menu edit/create


### Expected result AFTER applying this Pull Request
Schema tab in menu edit/create


### Link to documentations
Please select:
